invenio_app_ils/ui/backoffice/src/pages/UserDetails/state/reducer.js

```diff
@@ -0,0 +1,30 @@
+import { IS_LOADING, SUCCESS, HAS_ERROR } from './types';
+
+export const initialState = {
+  isLoading: true,
+  hasError: false,
+  data: {},
+};
+
+export default (state = initialState, action) => {
+  switch (action.type) {
+    case IS_LOADING:
+      return { ...state, isLoading: true };
+    case SUCCESS:
+      return {
+        ...state,
+        isLoading: false,
+        data: action.payload,
```

Comment:
I think `data` should be null as `hasError` is the responsible for the error state. Maybe I would add an `error` state if we want to show the errorred response. But I don't think that the response is quite explanatory in our case.

Reply:
Wait, what is happening here?
Until now, what has been done by @topless was to have:
```
isLoading: boolean
data: contains result payload on success or error payload on error
hasError: boolean
```
If we change it, we change it everywhere.
Correct me if I am wrong please.

Reply:
@ntarocco yes that was the flow till now, the component will enter error state or not according to `hasError` flag. An approach closer to what @zzacharo suggests would be to have 

```
{
  ...
  data: {title: '...', ...} or null if it has an error,
  error: {msg: '..', status:' ...'} or null if it doesn't have an error,
  ...
}
```
Like this we don't depend on Booleans to figure it out but the object themselves. I am absolutely fine with both or similar approaches.

Reply:
for now I kept it consistent with the rest of the cases, if we want to change it we should change it in all the places, shall we discuss ?